src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,62 @@
+package duke.task;
+
+/**
+ * A Task class describing the tasks and keeping track of the task status whether is done or not.
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Creates a task object
+     *
+     * @param description the data to be stored
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * The status icon X or no icon if the task is done or not done
+     *
+     * @return a string of "X" or " "
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    /**
+     * A method to mark the task as done
+     */
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * A method to unmark the task
+     */
+    public void unMark() {
```

Comment:
Maybe can change name to markAsNotDone to keep consistency with the previous method name?